.sandbox/provision/lifecycle-asg/common.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+: ${MAX_RETRIES:=30}
+: ${VOLUME_SIZE:=10}
+: ${DEVICE_NAME:="/dev/xvdf"}
+: ${SANDBOX_NAME_TAG:="SandboxManaged-SandboxName"}
+: ${SANDBOX_ID_TAG:="SandboxManaged-SandboxID"}
+: ${SANDBOX_ASG_TAG:="sandbox-asg"}
+
+function fatal() {
+    echo "$@" >&2
+    exit 1
+}
+
+function process_response() {
+    local response
+    if (( $# == 0 )) ; then
+        response="$(< /dev/stdin)"
+    else
+        response="$1"
+    fi
+
+    if [[ "$response" = "None" ]]; then
+        response=""
+    fi
+    echo "$response"
+}
+
+function get_volume_id() {
+    aws ec2 describe-volumes --filters Name=tag:"$SANDBOX_ID_TAG",Values="$SANDBOX_ID" --query 'Volumes[0].VolumeId' --output text | process_response
+}
+
+function create_volume_if_needed() {
+    local volume_id="$(get_volume_id)"
+    if [[ -z "$volume_id" ]]; then
+        volume_id="$(aws ec2 create-volume --size "$VOLUME_SIZE" --availability-zone "$AVAILABILITY_ZONE" --tag-specification "ResourceType=volume,Tags=[{Key=$SANDBOX_ID_TAG,Value="$SANDBOX_ID"}]" --query "VolumeId" --output text | process_response)"
+    fi
+    echo "$volume_id"
+}
+
+function delete_volume() {
+    local volume_id="$(get_volume_id)"
+    [[ -z "$volume_id" ]] || aws ec2 delete-volume --volume-id "$volume_id"
+}
+
+function get_instance_id() {
+    aws ec2 describe-instances --filters Name=tag:"$SANDBOX_ID_TAG",Values="$SANDBOX_ID" Name=instance-state-name,Values=running --query 'Reservations[0].Instances[0].InstanceId' --output text | process_response
+}
+
+function terminate_instance() {
+    local instance_id
+    instance_id="$(get_instance_id)"
+    [[ -z "$instance_id" ]] || {
+        aws ec2 terminate-instances --instance-ids "$instance_id"
+        aws ec2 wait instance-terminated --instance-ids "$instance_id"
```

Comment:
I think for all `aws ec2 wait`, we can all ignore the result, so `aws ec2 wait ... || true`